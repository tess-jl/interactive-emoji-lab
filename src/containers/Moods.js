import React, { useReducer } from 'react';
import taskReducer from '../reducers/taskReducer';
import { getFace } from '../selectors/moodSelectors';
import { DRINK_COFFEE, drinkCoffee, EAT_SNACK, eatSnack, TAKE_NAP, takeNap, STUDY, study } from '../actions/taskActions';
import Controls from '../components/controls/Controls';
import Face from '../components/face/Face';

const Moods = () => {
  const initialState = {
    coffees: 0, 
    snacks: 0,
    naps: 0,
    studies: 0
  };

  const [moods, dispatch] = useReducer(taskReducer, initialState);

  const face = getFace(moods);

  const actions = [
    { name: DRINK_COFFEE, text: 'Drink Coffee', stateName: 'coffees', count: moods.coffees, actionCreator: () => dispatch(drinkCoffee()) },
    { name: EAT_SNACK, text: 'Snack', stateName: 'snacks', count: moods.snacks, actionCreator: () => dispatch(eatSnack()) },
    { name: TAKE_NAP, text: 'Nap', stateName: 'naps', count: moods.naps, actionCreator: () => dispatch(takeNap()) },
    { name: STUDY, text: 'Study', stateName: 'studies', count: moods.studies, actionCreator: () => dispatch(study()) },
  ];

  return (
    <>
      <Controls actions={actions} />
      <Face emoji={face} />
    </>
  );
};

export default Moods;
