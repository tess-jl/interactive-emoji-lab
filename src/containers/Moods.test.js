import React from 'react';
import Moods from './Moods';
import { shallow } from 'enzyme';

describe('container', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<Moods />);
    expect(wrapper).toMatchSnapshot();
  });
});

