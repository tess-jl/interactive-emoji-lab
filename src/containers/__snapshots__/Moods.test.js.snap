// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`container matches a snapshot 1`] = `
<Fragment>
  <Controls
    actions={
      Array [
        Object {
          "actionCreator": [Function],
          "count": 0,
          "name": "DRINK_COFFEE",
          "stateName": "coffees",
          "text": "Drink Coffee",
        },
        Object {
          "actionCreator": [Function],
          "count": 0,
          "name": "EAT_SNACK",
          "stateName": "snacks",
          "text": "Snack",
        },
        Object {
          "actionCreator": [Function],
          "count": 0,
          "name": "TAKE_NAP",
          "stateName": "naps",
          "text": "Nap",
        },
        Object {
          "actionCreator": [Function],
          "count": 0,
          "name": "STUDY",
          "stateName": "studies",
          "text": "Study",
        },
      ]
    }
  />
  <Face
    emoji="ðŸ˜ "
  />
</Fragment>
`;
